pkgs/servers/klipper/default.nix

```diff
@@ -0,0 +1,50 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, python2
+}:
+stdenv.mkDerivation rec {
+  name = "klipper";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "KevinOConnor";
+    repo = "klipper";
+    rev = "v${version}";
+    sha256 = "1ijy2ij9yii5hms10914i614wkjpsy0k4rbgnm6l594gphivdfm7";
+  };
+
+  sourceRoot = "source/klippy";
+
+  # there is currently an attempt at moving it to Python 3, but it will remain
+  # Python 2 for the foreseeable future.
+  # c.f. https://github.com/KevinOConnor/klipper/pull/3278
+  nativeBuildInputs = [ (python2.withPackages (p: with p; [ cffi pyserial greenlet jinja2 ])) ];
+
+  # we need to run this to prebuild the chelper.
+  postBuild = "python2 ./chelper/__init__.py";
+
+  # NB: We don't move the main entry point into `/bin`, or even symlink it,
+  # because it uses relative paths to find necessary modules. We could wrap but
+  # this is used 99% of the time as a service, so it's not worth the effort.
+  installPhase = ''
+    mkdir -p $out/lib
+    cp -r ./* $out/lib
+
+    chmod 755 $out/lib/klippy.py
```

Comment:
This should probably go to a subdirectory.

Reply:
What do you mean?

Reply:
By FHS convention, we should not put stuff directly to `$out/lib`, other than libraries that stuff can link against. Private libraries and other similar things would go to `$pkglibdir` (`$prefix/lib/$project_name`).

Reply:
Understood, fixed!